client/asset/dcr/dcr.go

```diff
@@ -966,6 +1042,159 @@ func (dcr *ExchangeWallet) Balance() (*asset.Balance, error) {
 	return bal, nil
 }
 
+// Balance should return the total available funds in the wallet.
+func (dcr *ExchangeWallet) Balance() (*asset.Balance, error) {
+	bal, err := dcr.balance()
+	if err != nil {
+		return nil, err
+	}
+
+	reserves := dcr.reserves()
+	if reserves > bal.Available { // unmixed (immature) probably needs to trickle in
+		dcr.log.Warnf("Available balance is below configured reserves: %f < %f",
+			toDCR(bal.Available), toDCR(reserves))
+		reserves = bal.Available
+	}
+	bal.Available -= reserves
+	bal.Locked += reserves
+
+	return bal, nil
+}
+
+func bondsFeeBuffer(highFeeRate uint64) uint64 {
+	const inputCount uint64 = 20 // plan for lots of inputs
```

Comment:
Probably true in most cases, esp. with the high fee rate and the "tracks" multiplier below, but this helper is in `client/asset/dcr` so other assets would do differently.  I expect eth would be quite a bit simpler in this regard (no UTXO business to worry about), although this could be a lot of value for btc.